src/Symfony/Component/Mailer/Bridge/Postal/Transport/PostalApiTransport.php

```diff
@@ -0,0 +1,154 @@
+            $result = $response->toArray(false);
+        } catch (DecodingExceptionInterface $e) {
+            throw new HttpTransportException('Unable to send an email: '.$response->getContent(false).\sprintf(' (code %d).', $statusCode), $response);
+        } catch (TransportExceptionInterface $e) {
+            throw new HttpTransportException('Could not reach the remote Postal server.', $response, 0, $e);
+        }
+
+        if (200 !== $statusCode) {
+            throw new HttpTransportException('Unable to send an email: '.$result['message'].\sprintf(' (code %d).', $statusCode), $response);
+        }
+
+        $sentMessage->setMessageId($result['message_id']);
+
+        return $response;
+    }
+
+    private function getPayload(Email $email, Envelope $envelope): array
+    {
+        $payload = [
+            'from' => $this->formatAddress($envelope->getSender()),
+            'to' => $this->formatAddresses($this->getRecipients($email, $envelope)),
+            'subject' => $email->getSubject(),
+        ];
+        if ($emails = $email->getCc()) {
+            $payload['cc'] = $this->formatAddresses($emails);
+        }
+        if ($emails = $email->getBcc()) {
+            $payload['bcc'] = $this->formatAddresses($emails);
+        }
+        if ($email->getTextBody()) {
+            $payload['plain_body'] = $email->getTextBody();
+        }
+        if ($email->getHtmlBody()) {
+            $payload['html_body'] = $email->getHtmlBody();
+        }
+        if ($attachments = $this->prepareAttachments($email)) {
+            $payload['attachments'] = $attachments;
+        }
+        if ($headers = $this->getCustomHeaders($email)) {
+            $payload['headers'] = $headers;
+        }
+        if ($emails = $email->getReplyTo()) {
+            $payload['reply_to'] = $this->formatAddress($emails[0]);
+        }
+
+        return $payload;
+    }
+
+    private function prepareAttachments(Email $email): array
+    {
+        $attachments = [];
+        foreach ($email->getAttachments() as $attachment) {
+            $attachments[] = [
+                'name' => $attachment->getFilename(),
+                'content_type' => $attachment->getContentType(),
+                'data' => base64_encode($attachment->getBody()),
+            ];
+        }
+
+        return $attachments;
+    }
+
+    private function getCustomHeaders(Email $email): array
+    {
+        $headers = [];
+        $headersToBypass = ['from', 'to', 'cc', 'bcc', 'subject', 'content-type', 'sender', 'reply-to'];
+        foreach ($email->getHeaders()->all() as $name => $header) {
+            if (\in_array($name, $headersToBypass, true)) {
+                continue;
+            }
+
+            $headers[] = [
+                'key' => $header->getName(),
+                'value' => $header->getBodyAsString(),
+            ];
+        }
+
+        return $headers;
+    }
+
+    private function formatAddress(Address $address): string
```

Comment:
IMO we should inline this method (the name is misleading to me as we do not format anything here)